Automattic/vip-workflow-plugin
# Change email notifications to not go to blog admins

## Description

This tweaks the notifications system to be off by default, until an email address and/or webhook notification is configured. This way, blog admins are not always notified as that doesn't quite make sense. The recipients can be modified via filters, until we support this via the custom status.

I have also modified the whole settings configuration to show two textboxes that can be tweaked, rather than turning it on/off and then configuring it. That feels closer to how WordPress does it in the settings page.

![Screenshot 2024-09-16 at 3 54 31 PM](https://github.com/user-attachments/assets/83d1a322-c040-485a-b298-53eaef8291f6)

In addition, the email body has been tweaked to be a lot simpler. I tried to get it closer to the VIP notifications way of doing it without changing it to an HTML email body. This is what it looks like:

![Screenshot 2024-09-17 at 3 35 14 PM](https://github.com/user-attachments/assets/5c67432c-9bf6-4442-8ede-4de8deae16b6)

It's simple, and maybe HTML does make sense but figured for how now this would be better. I haven't incorporated a filter in, as we would need to allow changing the subject/body/headers. So for now it's just a simple one.

## Steps to Test

- Ensure mailpit and cron are configured on `dev-env`
- Configure a dummy email and a webhook
- Create a new post, and publish it
- Go to your mailpit url, and ensures the emails show up.
